Pinta.Core/Classes/Document.cs

```diff
@@ -54,11 +54,11 @@ public DocumentSelection Selection {
 		}
 	}
 
-	public DocumentSelection PreviousSelection { get; set; } = new ();
+	public DocumentSelection PreviousSelection { get; set; } = new (PintaCore.Workspace);
 
 	public Document (Size size)
 	{
-		Selection = new DocumentSelection ();
```

Comment:
Rather than passing in `PintaCore.Workspace`, I think it would work to just pass in a reference to the owning document? 
DocumentSelection only uses the `ActiveDocument` of the workspace, which is really just this document